pkg/standalone-utils/test/GaugeActions.test.ts

```diff
@@ -0,0 +1,659 @@
+    });
+    gaugeFactory = await deploy('v2-liquidity-mining/LiquidityGaugeFactory', { args: [gaugeImplementation.address] });
+
+    gauge = await deployedAt(
+      'v2-liquidity-mining/LiquidityGaugeV5',
+      await deployGauge(gaugeFactory, lpToken.address) // No weight cap.
+    );
+
+    // Type weight is ignored in the mock controller.
+    await gaugeController.add_type('Ethereum', 0);
+    await gaugeController.add_gauge(gauge.address, 0); // Type: Ethereum in mock controller.
+  });
+
+  describe('gaugeDeposit', () => {
+    let tokenSender: Account, tokenRecipient: Account;
+
+    context('when using relayer library directly', () => {
+      it('reverts', async () => {
+        expect(
+          relayerLibrary.connect(userSender).gaugeDeposit(gauge.address, userSender.address, gauge.address, fp(1))
+        ).to.be.revertedWith('Incorrect sender');
+      });
+    });
+
+    context('when caller != sender and sender != relayer', () => {
+      it('reverts', async () => {
+        expect(
+          relayerLibrary.connect(userSender).gaugeDeposit(gauge.address, other.address, gauge.address, fp(1))
+        ).to.be.revertedWith('Incorrect sender');
+      });
+    });
+
+    context('when sender does not have enough BPT', () => {
+      it('reverts', async () => {
+        expect(
+          relayer
+            .connect(userSender)
+            .multicall([
+              encodeDeposit({ gauge: gauge, sender: userSender, recipient: userSender, amount: totalLpTokens.add(1) }),
+            ])
+        ).to.be.reverted;
+      });
+    });
+
+    context('when sender has enough BPT', () => {
+      context('sender = senderUser, recipient = senderUser', () => {
+        sharedBeforeEach('', async () => {
+          tokenSender = userSender;
+          tokenRecipient = userSender;
+        });
+
+        itDepositsWithEnoughTokens();
+      });
+
+      context('sender = senderUser, recipient = relayer', () => {
+        sharedBeforeEach('', async () => {
+          tokenSender = userSender;
+          tokenRecipient = relayer;
+        });
+
+        itDepositsWithEnoughTokens();
+      });
+
+      context('sender = relayer, recipient = senderUser', () => {
+        sharedBeforeEach('', async () => {
+          await lpToken.connect(userSender).transfer(relayer.address, totalLpTokens);
+          tokenSender = relayer;
+          tokenRecipient = userSender;
+        });
+
+        itDepositsWithEnoughTokens();
+      });
+
+      context('sender = relayer, recipient = relayer', () => {
+        sharedBeforeEach('', async () => {
+          await lpToken.connect(userSender).transfer(relayer.address, totalLpTokens);
+          tokenSender = relayer;
+          tokenRecipient = relayer;
+        });
+
+        itDepositsWithEnoughTokens();
+      });
+    });
+
+    function itDepositsWithEnoughTokens() {
+      context('when depositing some of the tokens', () => {
+        itDepositsWithRefsAndAmounts(totalLpTokens.div(3));
+      });
```

Comment:
If you're doing this, might as all also add a test for `total.add(1)` :)

Reply:
Could be, but that's already covered in a context above (`when sender does not have enough BPT`)